ros/src/actuation/vehicles/packages/as/nodes/ssc_interface/ssc_interface.cpp

```diff
@@ -0,0 +1,286 @@
+  private_nh_.param<double>("acceleration_limit", acceleration_limit_, 3.0);
+  private_nh_.param<double>("deceleration_limit", deceleration_limit_, 3.0);
+  private_nh_.param<double>("max_curvature_rate", max_curvature_rate_, 0.15);
+  private_nh_.param<double>("agr_coef_a", agr_coef_a_, 15.713);
+  private_nh_.param<double>("agr_coef_b", agr_coef_b_, 0.053);
+  private_nh_.param<double>("agr_coef_c", agr_coef_c_, 0.042);
+
+  rate_ = new ros::Rate(loop_rate_);
+
+  // subscribers from autoware
+  vehicle_cmd_sub_ = nh_.subscribe("vehicle_cmd", 1, &SSCInterface::callbackFromVehicleCmd, this);
+  engage_sub_ = nh_.subscribe("vehicle/engage", 1, &SSCInterface::callbackFromEngage, this);
+
+  // subscribers from SSC
+  module_states_sub_ = nh_.subscribe("as/module_states", 1, &SSCInterface::callbackFromSSCModuleStates, this);
+  curvature_feedback_sub_ =
+      new message_filters::Subscriber<automotive_platform_msgs::CurvatureFeedback>(nh_, "as/curvature_feedback", 10);
+  throttle_feedback_sub_ =
+      new message_filters::Subscriber<automotive_platform_msgs::ThrottleFeedback>(nh_, "as/throttle_feedback", 10);
+  brake_feedback_sub_ =
+      new message_filters::Subscriber<automotive_platform_msgs::BrakeFeedback>(nh_, "as/brake_feedback", 10);
+  gear_feedback_sub_ =
+      new message_filters::Subscriber<automotive_platform_msgs::GearFeedback>(nh_, "as/gear_feedback", 10);
+  velocity_accel_sub_ =
+      new message_filters::Subscriber<automotive_platform_msgs::VelocityAccel>(nh_, "as/velocity_accel", 10);
+  // subscribers from PACMod
+  wheel_speed_sub_ =
+      new message_filters::Subscriber<pacmod_msgs::WheelSpeedRpt>(nh_, "pacmod/parsed_tx/wheel_speed_rpt", 10);
+  steering_wheel_sub_ =
+      new message_filters::Subscriber<pacmod_msgs::SystemRptFloat>(nh_, "pacmod/parsed_tx/steer_rpt", 10);
+  ssc_feedbacks_sync_ = new message_filters::Synchronizer<SSCFeedbacksSyncPolicy>(
+      SSCFeedbacksSyncPolicy(10), *velocity_accel_sub_, *curvature_feedback_sub_, *throttle_feedback_sub_,
+      *brake_feedback_sub_, *gear_feedback_sub_, *wheel_speed_sub_, *steering_wheel_sub_);
+  ssc_feedbacks_sync_->registerCallback(
+      boost::bind(&SSCInterface::callbackFromSSCFeedbacks, this, _1, _2, _3, _4, _5, _6, _7));
+
+  // publishers to autoware
+  vehicle_status_pub_ = nh_.advertise<autoware_msgs::VehicleStatus>("vehicle_status", 10);
+  current_twist_pub_ = nh_.advertise<geometry_msgs::TwistStamped>("vehicle/twist", 10);
+
+  // publishers to SSC
+  steer_mode_pub_ = nh_.advertise<automotive_platform_msgs::SteerMode>("as/arbitrated_steering_commands", 10);
+  speed_mode_pub_ = nh_.advertise<automotive_platform_msgs::SpeedMode>("as/arbitrated_speed_commands", 10);
+  turn_signal_pub_ = nh_.advertise<automotive_platform_msgs::TurnSignalCommand>("as/turn_signal_command", 10);
+  gear_pub_ = nh_.advertise<automotive_platform_msgs::GearCommand>("as/gear_select", 1, true);
+}
+
+SSCInterface::~SSCInterface()
+{
```

Comment:
Delete pointers initialized in the constructor, or do not use pointers.

Reply:
Since support for Ubuntu 14.04/ROS Indigo is no longer required, we should be able to support C++11 features like using a smart pointer here for memory management, correct?